tests/engine/test_calc_job/test_multi_codes_run_parallel_False_.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+exec > _scheduler-stdout.txt
+exec 2> _scheduler-stderr.txt
+
+
+'/bin/bash'
+
+'mpirun' '-np' '1' '/bin/bash'
```

Comment:
Mmm this is strange no? I expected `content = content.rstrip() + '\n'` to add a final newline in the file but apparently it didn't, it just ended the current line. More surprisingly so: the linter didn't complain that there was no newline end?

I mean, if it is working we can leave it like this, it is just...unexpected.

Reply:
It is expected though. The `rstrip` strips _all_ whitespace, so also the newline character, which you then re-append. However, having a terminating newline character, does not show up visually as an empty line. It just means the line is terminated and the carriage return goes to the _start_ of the next line. If you want to see a visual empty line, you would need another new line character. Note that Github visualizes this difference (in diffs where this changes) with a symbol at the end of the file, saying "no newline character" or something to that effect.

Reply:
Ah, yeah, for the `rstrip()` I did some tests in the shell and printing so I didn't know exactly how it would behave with write, there my "I expected..." meant "I intended that...". The unexpected part is more about the linter not complaining.